schemas/com.snowplowanalytics.monitor/application/jsonschema/1-0-0

```diff
@@ -0,0 +1,112 @@
+{
+  "$schema": "http://iglucentral.com/schemas/com.snowplowanalytics.self-desc/schema/jsonschema/1-0-0#",
+  "description": "Schema for an application entity, attached as context to application status report events",
+  "self": {
+    "vendor": "com.snowplowanalytics.monitor",
+    "name": "application",
+    "format": "jsonschema",
+    "version": "1-0-0"
+  },
+
+  "type": "object",
+  "properties": {
+    "name": {
+      "description": "The name of the application, eg rdb-loader",
+      "type": ["string", "null"],
+      "maxLength": 256
+    },
+    "category": {
+      "description": "The category of the application, eg loader",
+      "type": ["string", "null"],
+      "maxLength": 256
+    },
+    "version": {
+      "description": "The version of the application, eg 1.22.1",
+      "type": ["string", "null"],
+      "maxLength": 32
+    },
+    "instance_id": {
```

Comment:
instance_id is a bit overloaded from AWS as the id of an EC2 "instance" - can this be just "id" and for "instance_name" just "name"?

Reply:
`instance_id` can be just `id` but we already have a `name` property:

```
"name": {
      "description": "The name of the application, eg rdb-loader",
      (...)
    },
"instance_name": {
      "description": "The name of the application instance, eg com-acme-prod1-rdb-loader-1",
      (...)
    }
```

I guess we can get rid of `name=rdb-loader` if it will always be clear what application that is from the (currently so called) `instance_name`?

Reply:
Maybe name becomes app_name and instance_name becomes name?

Reply:
I think it helps to look at the fully qualified names:

```
com.snowplowanalytics.monitor.application.name
com.snowplowanalytics.monitor.application.instanceName
```

vs 

```
com.snowplowanalytics.monitor.application.appName
com.snowplowanalytics.monitor.application.name
```

I think what we have now works better that way. WDYT?

(Note to self: There are a few rogue snake_cases in these schemas.)